src/preset_cli/cli/superset/sync/dbt/datasets.py

```diff
@@ -70,6 +70,19 @@ def create_dataset(
     Virtual datasets are created when the table database is different from the main
     database, for systems that support cross-database queries (Trino, BigQuery, etc.)
     """
+    kwargs = {
+        "database": database["id"],
+        "catalog": model["database"],
+        "schema": model["schema"],
+        "table_name": model.get("alias") or model["name"],
+    }
+    try:
+        # try to create dataset with catalog
+        return client.create_dataset(**kwargs)
+    except SupersetError as ex:
+        if not no_catalog_support(ex):
+            raise ex
+
     url = make_url(database["sqlalchemy_uri"])
     if model_in_database(model, url):
```

Comment:
... then avoid this `if/else` and have a single `if not model_in_database(model, url)`